source/main.cpp

```diff
@@ -260,6 +270,33 @@ void onMinuteOffsetChanged(ConfigItemIntegerRange *item, int32_t offset)
     offsetMinutes = offset;
 }
 
+static void setTimeInSettings() {
+    OSTime ot;
+    OSCalendarTime ct;
+    char timeString[256];
+
+    ot = NTPGetTime(NTP_SERVER);
+    if (ot == 0) {
+        WUPSConfigItem_SetDisplayName(ntpTimeHandle->handle, "Current NTP Time: N/A");
+    } else {
+        OSTicksToCalendarTime(ot, &ct);
+        snprintf(timeString, 255, "Current NTP Time: %04d-%02d-%02d %02d:%02d:%02d:%04d:%04d\n", ct.tm_year, ct.tm_mon + 1, ct.tm_mday, ct.tm_hour, ct.tm_min, ct.tm_sec, ct.tm_msec, ct.tm_usec);
+        WUPSConfigItem_SetDisplayName(ntpTimeHandle->handle, timeString);
+    }
+
+    ot = OSGetTime();
+    OSTicksToCalendarTime(ot, &ct);
+    snprintf(timeString, 255, "Current SYS Time: %04d-%02d-%02d %02d:%02d:%02d:%04d:%04d\n", ct.tm_year, ct.tm_mon + 1, ct.tm_mday, ct.tm_hour, ct.tm_min, ct.tm_sec, ct.tm_msec, ct.tm_usec);
+    WUPSConfigItem_SetDisplayName(sysTimeHandle->handle, timeString);
+}
+
+static void settingsThreadMain() {
+    while(settingsThreadActive) {
+        setTimeInSettings();
+        sleep(1);
+    }
+}
```

Comment:
This seems like it'll fetch the time from the NTP server every second. Could it be better to fetch the time once, and store it as an offset to what the system time is?

Reply:
1sec might be overkill but the reason for this thread is to show whatever the NTP server returned when the last systemtime was grabbed. So I think it would be nice to get a new NTP time each X seconds buf ofc. that's up to you.